inference-engine/tests/unit/cpu/mkldnn_memory_desc_test.cpp

```diff
@@ -45,9 +45,1126 @@ TEST(MemDescTest, Conversion) {
+    Shape dynamicShape(ngraph::PartialShape{{16}, {7, 15}, {-1, -1}, {-1, -1}});
+    Shape staticShape(SizeVector{1, 1, 10, 10});
+
+    auto isDefinedStaticShape = [&] (dnnl::memory::format_tag fmt) {
+        DnnlBlockedMemoryDescPtr dnnl_blk_desc_ptr;
+        // DnnlBlockedMemoryDesc
+        dnnl_blk_desc_ptr = std::make_shared<DnnlBlockedMemoryDesc>(staticShape, mkldnn::memory::data_type::u8, fmt);
+        ASSERT_TRUE(dnnl_blk_desc_ptr->isDefined());
+
+        // CpuBlockedMemoryDesc
+        CpuBlockedMemoryDescPtr cpu_blk_desc_ptr = std::make_shared<CpuBlockedMemoryDesc>(dnnl_blk_desc_ptr->getPrecision(), dnnl_blk_desc_ptr->getShape(),
+                                                                                          dnnl_blk_desc_ptr->getBlockDims(), dnnl_blk_desc_ptr->getOrder(),
+                                                                                          dnnl_blk_desc_ptr->getOffsetPadding(),
+                                                                                          dnnl_blk_desc_ptr->getOffsetPaddingToData(),
+                                                                                          dnnl_blk_desc_ptr->getStrides());
+        ASSERT_TRUE(cpu_blk_desc_ptr->isDefined());
+    };
+
+    auto isDefinedDynamicShape = [&] (dnnl::memory::format_tag fmt) {
+        DnnlBlockedMemoryDescPtr dnnl_blk_desc_ptr;
+        // DnnlBlockedMemoryDesc
+        dnnl_blk_desc_ptr = std::make_shared<DnnlBlockedMemoryDesc>(dynamicShape, mkldnn::memory::data_type::u8, fmt);
+        ASSERT_FALSE(dnnl_blk_desc_ptr->isDefined());
+
+        // CpuBlockedMemoryDesc
+        CpuBlockedMemoryDescPtr cpu_blk_desc_ptr = std::make_shared<CpuBlockedMemoryDesc>(dnnl_blk_desc_ptr->getPrecision(), dnnl_blk_desc_ptr->getShape(),
+                                                                                          dnnl_blk_desc_ptr->getBlockDims(), dnnl_blk_desc_ptr->getOrder(),
+                                                                                          dnnl_blk_desc_ptr->getOffsetPadding(),
+                                                                                          dnnl_blk_desc_ptr->getOffsetPaddingToData(),
+                                                                                          dnnl_blk_desc_ptr->getStrides());
+        ASSERT_FALSE(cpu_blk_desc_ptr->isDefined());
+    };
+
+    dnnl::memory::format_tag payload[] {
+            dnnl::memory::format_tag::nChw16c,
+            dnnl::memory::format_tag::nhwc,
+            dnnl::memory::format_tag::nchw,
+            dnnl::memory::format_tag::NChw16n16c,
+            dnnl::memory::format_tag::BAcd16a16b,
+            dnnl::memory::format_tag::Acdb16a,
+    };
+
+    for (const auto &p : payload) {
+        isDefinedStaticShape(p);
+        isDefinedDynamicShape(p);
+    }
+}
+
+TEST(MemDescTest, getMaxMemSize) {
+    Shape dynamicShape(ngraph::PartialShape{{16}, {7, 15}, {-1, -1}, {-1, -1}});
```

Comment:
add test cases for dynamic shapes with defined upper bound

Reply:
looks like you can use the same scheme as in https://github.com/openvinotoolkit/openvino/pull/7431#discussion_r708308944
for `getMaxMemSize` `getCurrentMemSize`